mlflow/utils/autologging/__init__.py

```diff
@@ -1023,219 +1048,207 @@ def safe_patch_function(*args, **kwargs):
-                if failed_during_original or _is_testing():
-                    raise
-
-                try_log_autologging_event(
-                    AutologgingEventLogger.get_logger().log_patch_function_error,
-                    session,
-                    destination,
-                    function_name,
-                    args,
-                    kwargs,
-                    e,
-                )
+                    try_log_autologging_event(
+                        AutologgingEventLogger.get_logger().log_patch_function_error,
+                        session,
+                        destination,
+                        function_name,
+                        args,
+                        kwargs,
+                        e,
+                    )
 
-                _logger.warning(
-                    "Encountered unexpected error during %s autologging: %s",
-                    autologging_integration,
-                    e,
-                )
+                    _logger.warning(
+                        "Encountered unexpected error during %s autologging: %s",
+                        autologging_integration,
+                        e,
+                    )
 
-            if _is_testing() and not preexisting_run_for_testing:
-                # If an MLflow run was created during the execution of patch code, verify that
-                # it is no longer active and that it contains expected autologging tags
-                assert not mlflow.active_run(), (
-                    "Autologging integration %s leaked an active run" % autologging_integration
-                )
-                if patch_function_run_for_testing:
-                    _validate_autologging_run(
-                        autologging_integration, patch_function_run_for_testing.info.run_id
+                if _is_testing() and not preexisting_run_for_testing:
+                    # If an MLflow run was created during the execution of patch code, verify that
+                    # it is no longer active and that it contains expected autologging tags
+                    assert not mlflow.active_run(), (
+                        "Autologging integration %s leaked an active run" % autologging_integration
                     )
+                    if patch_function_run_for_testing:
+                        _validate_autologging_run(
+                            autologging_integration, patch_function_run_for_testing.info.run_id
+                        )
 
-            if original_has_been_called:
-                return original_result
-            else:
-                return original(*args, **kwargs)
+                if original_has_been_called:
+                    return original_result
+                else:
+                    return original(*args, **kwargs)
 
     _wrap_patch(destination, function_name, safe_patch_function)
 
 
-@contextmanager
-def _augment_mlflow_warnings(autologging_integration):
```

Comment:
Replaced with a `reroute_warnings` on option on each of `set_mlflow_events_and_warnings_behavior_globally` / `set_non_mlflow_warnings_behavior_for_current_thread` (introduced in `mlflow/utils/autologging/logging_and_warnings.py`)